### PR TITLE
Hs 415 saving phone no on db

### PR DESCRIPTION
**Pull request by dlvinh**

### CODE REVIEW

File: lib/data/data.dart [hunk 1]
```diff
@@ -54,20 +54,24 @@ enum Roles {
...
+
+  static PhoneCode fromCodeString(String stringCode) =>
+      values.firstWhere((element) => element.country == stringCode,
+          orElse: () => invalid);
 }
 
 class PhoneNumber {
   final String _countryCodeKey = 'countryCode';
   final String _numberKey = 'numberKey';
-  CountryCallingCode countryCode;
```
Comment (suesitran) [inline lib/data/data.dart:70]: can it be final?

File: lib/data/data.dart [hunk 2]
```diff
@@ -54,20 +54,24 @@ enum Roles {
...
+  PhoneCode countryCode;
   String phoneNumber;
```
Comment (suesitran) [inline lib/data/data.dart:75]: can it be final?

File: lib/features/booking/view_model/cubit/add_inspection_booking_cubit.dart [hunk 3]
```diff
@@ -20,7 +20,7 @@ class AddInspectionBookingCubit extends Cubit<AddInspectionBookingState> {
     try {
       UserDetail user = await authenticationService.getCurrentUser();
       List<Roles> userRole = await storageService.member.getUserRoles(user.uid);
-      final String? phoneNumber =
+      PhoneNumber? phoneNumber =
```
Comment (suesitran) [inline lib/features/booking/view_model/cubit/add_inspection_booking_cubit.dart:23]: can this be final?

File: lib/features/booking/view_model/cubit/add_inspection_booking_cubit.dart [hunk 4]
```diff
@@ -90,6 +90,19 @@ class AddInspectionBookingCubit extends Cubit<AddInspectionBookingState> {
     }
   }
 
+  void updateProfilePhoneNumber(String phoneNo,
+      {PhoneCode countryCode = PhoneCode.au}) async {
+    try {
+      UserDetail user = await authenticationService.getCurrentUser();
```
Comment (suesitran) [inline lib/features/booking/view_model/cubit/add_inspection_booking_cubit.dart:96]: this also can be final

File: lib/features/booking/view_model/cubit/add_inspection_booking_cubit.dart [hunk 5]
```diff
@@ -90,6 +90,19 @@ class AddInspectionBookingCubit extends Cubit<AddInspectionBookingState> {
...
+      final String formatNumber = phoneNo.substring(1); // remove first zero
```
Comment (suesitran) [inline lib/features/booking/view_model/cubit/add_inspection_booking_cubit.dart:97]: what if user does not enter the first number as 0?

**Review by suesitran — CHANGES_REQUESTED** [review]

File: lib/features/booking/view_model/cubit/add_inspection_booking_cubit.dart (hunk 5 above)
Comment (dlvinh) [reply lib/features/booking/view_model/cubit/add_inspection_booking_cubit.dart:97]: Actually, validation has been implemented on UI part. Phone number should match with area code ( always have zero atm)

File: lib/data/data.dart (hunk 2 above)
Comment (dlvinh) [reply lib/data/data.dart:75]: already add final

File: lib/data/data.dart (hunk 1 above)
Comment (dlvinh) [reply lib/data/data.dart:70]: already add final

File: lib/data/data.dart [hunk 6]
```diff
@@ -54,21 +54,25 @@ enum Roles {
...
-enum CountryCallingCode {
+enum PhoneCode {
   //TODO: add more value for upcomming story
   au('+61'),
   invalid('invalid');
 
-  const CountryCallingCode(this.country);
+  const PhoneCode(this.country);
   final String country;
   String get code => country;
```
[the remark this answers is not in the document]
Comment (dlvinh) [reply lib/data/data.dart:64]: will fix it in other PR

**Pull request merged**